Api changes

nadeemkhedr opened the pull request:
- Add server side validation
- Add timestamp fields for opportunities (used in dashboard)
- All data returned are json api compliant

kamilogorek commented on models/opportunity.js:
```diff
@@ -2,13 +2,29 @@
+    },
+    dateFrom: {
+        type: Date,
+        validate: [dateValidator, 'Date to must be greater than Date from']
```
There's great module from `hapijs` creators called `joy` for performing object validations as well – https://github.com/hapijs/joi
However feel free to use mongoose validate functions if you prefer so :)

kamilogorek commented on models/opportunity.js:
```diff
@@ -2,13 +2,29 @@
+//add createdAt, updatedAt
+OpportunitySchema.plugin(timestamps);
+
+function dateValidator(value) {
```
I'd use more descriptive name, as `dateValidator` is very vague. Something like `earlierThanDateTo` would work much nicer imo.

kamilogorek commented on routes/developers.js:
```diff
@@ -5,31 +5,31 @@ const developersController = require('../controllers/developers');
 module.exports = (server) => {
```
Those routes should stay as they were, but it should be registered as a plugin inside `index` instead and using `prefix` attribute.

See: http://hapijs.com/tutorials/plugins#plugin-options

nadeemkhedr replied:
done!

nadeemkhedr commented:
@kamilogorek the reason I added empty callback because its required, I changed that to `console.error` the error message, with a todo comment to log that when we have a logging system

Note: I am not writing any handling code, because the error should not happen, and honestly don't know how to handle it, because we are using internal hapi prefex as a plugin, so most likely if that error occurs it would be dependencies issue not user request issue

kamilogorek commented:
It is? Then they have an error in their docs :)

Merging
